src/scripts/cleanup.sh

```diff
@@ -0,0 +1,219 @@
+echo "success" > /tmp/build_status
+# should the cache be cleaned?
+
+GRADLE_DIRECTORY="$HOME/.gradle"
+GRADLE_CACHE_DIRECTORY="$GRADLE_DIRECTORY/caches"
+if [[ ! -e $GRADLE_CACHE_DIRECTORY ]]; then
+  echo "No cache directory, so no cleanup needed"
+  exit 0
+fi
+
+if [ -f "$GRADLE_CACHE_DIRECTORY/last_success_hash" ]; then
+  if cmp -s "$GRADLE_CACHE_DIRECTORY/last_success_hash" /tmp/git_last_hash ; then
+    echo "Cache does not need cleanup"
+    exit 0
+  fi
+fi
+
+if [ -n "${PARAM_APP_DIRECTORY}" ]; then
+  cd "$PARAM_APP_DIRECTORY" || exit 1
+fi
+
+# is this gradle 8+?
+gradleWrapperMainVersion="$(cat gradle/wrapper/gradle-wrapper.properties | grep distributionUrl | cut -d'-' -f 2 | cut -d'.' -f 1)"
+if [ "$gradleWrapperMainVersion" -ge "8" ]; then
+    # make it so the built-in GC runs
+    # for debugging
+    du -h --max-depth=1 "$GRADLE_CACHE_DIRECTORY"
+
+    echo "Clean cache for gradle >= 8"
+    # https://docs.gradle.org/8.0-rc-3/userguide/directory_layout.html#dir:gradle_user_home:configure_cache_cleanup
+    # https://docs.gradle.org/current/userguide/init_scripts.html#sec:using_an_init_script
+    GRADLE_INIT_DIRECTORY="$GRADLE_DIRECTORY/init.d"
+    if [[ ! -e $GRADLE_INIT_DIRECTORY ]]; then
+      mkdir -p $GRADLE_INIT_DIRECTORY
+    fi
+    echo -e "beforeSettings { settings -> settings.caches {\ndownloadedResources.removeUnusedEntriesAfterDays = 1\nreleasedWrappers.removeUnusedEntriesAfterDays = 1\nsnapshotWrappers.removeUnusedEntriesAfterDays = 1\ncreatedResources.removeUnusedEntriesAfterDays = 1\ncleanup = Cleanup.ALWAYS\n}}" > $GRADLE_INIT_DIRECTORY/cache-settings.gradle
+
+    touch /tmp/settings.gradle
+    cat > /tmp/cleanup.gradle << 'endmsg'
+task dummy {
+    group 'Dummy task triggering cleanup'
+    description 'Tasks which triggers dependency cleanup'
+    doLast {
+        println 'Dummy task execution'
+    }
+}
+endmsg
+    echo "A new cache entry will be created, cleaning files not accessed during the last 24 hours.."
+    du -h --max-depth=1 "$GRADLE_CACHE_DIRECTORY"
+    ./gradlew --stop
+    ./gradlew -b /tmp/cleanup.gradle dummy --no-daemon --info
+    # for debugging
+    du -h --max-depth=1 "$GRADLE_CACHE_DIRECTORY"
+    exit 0
+fi
+# this is the first successful build with this particular set of build files
+echo "Clean up cache for gradle < 8"
+touch /tmp/settings.gradle
+cat > /tmp/cleanup.gradle << 'endmsg'
```

Comment:
perhaps cleanup.gradle should be added as a separate file?

Reply:
Yes, but it is not possible to include files in the orb (technically all imports are embedded).